homeassistant/components/solarlog/const.py

```diff
@@ -17,83 +30,207 @@
-SENSOR_TYPES = {
-    "time": ["TIME", "last update", None, "mdi:calendar-clock"],
-    "power_ac": ["powerAC", "power AC", POWER_WATT, "mdi:solar-power"],
-    "power_dc": ["powerDC", "power DC", POWER_WATT, "mdi:solar-power"],
-    "voltage_ac": ["voltageAC", "voltage AC", ELECTRIC_POTENTIAL_VOLT, "mdi:flash"],
-    "voltage_dc": ["voltageDC", "voltage DC", ELECTRIC_POTENTIAL_VOLT, "mdi:flash"],
-    "yield_day": ["yieldDAY", "yield day", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
-    "yield_yesterday": [
-        "yieldYESTERDAY",
-        "yield yesterday",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "yield_month": [
-        "yieldMONTH",
-        "yield month",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "yield_year": ["yieldYEAR", "yield year", ENERGY_KILO_WATT_HOUR, "mdi:solar-power"],
-    "yield_total": [
-        "yieldTOTAL",
-        "yield total",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:solar-power",
-    ],
-    "consumption_ac": ["consumptionAC", "consumption AC", POWER_WATT, "mdi:power-plug"],
-    "consumption_day": [
-        "consumptionDAY",
-        "consumption day",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_yesterday": [
-        "consumptionYESTERDAY",
-        "consumption yesterday",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_month": [
-        "consumptionMONTH",
-        "consumption month",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_year": [
-        "consumptionYEAR",
-        "consumption year",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "consumption_total": [
-        "consumptionTOTAL",
-        "consumption total",
-        ENERGY_KILO_WATT_HOUR,
-        "mdi:power-plug",
-    ],
-    "total_power": ["totalPOWER", "total power", "Wp", "mdi:solar-power"],
-    "alternator_loss": [
-        "alternatorLOSS",
-        "alternator loss",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "capacity": ["CAPACITY", "capacity", PERCENTAGE, "mdi:solar-power"],
-    "efficiency": [
-        "EFFICIENCY",
-        "efficiency",
-        f"% {POWER_WATT}/{POWER_WATT}p",
-        "mdi:solar-power",
-    ],
-    "power_available": [
-        "powerAVAILABLE",
-        "power available",
-        POWER_WATT,
-        "mdi:solar-power",
-    ],
-    "usage": ["USAGE", "usage", None, "mdi:solar-power"],
-}
+
+@dataclass
+class SolarlogRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    json_key: str
+
+
+@dataclass
+class SolarLogSensorEntityDescription(
+    SensorEntityDescription, SolarlogRequiredKeysMixin
+):
+    """Describes Solarlog sensor entity."""
+
+
+SENSOR_TYPES: tuple[SolarlogSensorEntityDescription, ...] = (
+    SolarLogSensorEntityDescription(
+        key="time",
+        json_key="TIME",
+        name="last update",
+        icon="mdi:calendar-clock",
```

Comment:
```suggestion
        icon="mdi:calendar-clock",
```
`ìcon` should not be set when `device_class` is defined.

Reply:
Removed them, except for the solar sensors. I think it would make sense to have different icons for consumption and for production. Is that ok?

Reply:
Use `device_class` or `icon`, not both

Reply:
But with only an icon, will the new energy feature still work? I though the device class needed to be set.

Reply:
Tested it, works fine without device class. Fixed the remaining sensors by removing the device class or icon.